### PR TITLE
IGNITE-17249 DefaultMessagingService incorrect message sending order

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-17249

### CODE REVIEW

File: modules/network/src/main/java/org/apache/ignite/internal/network/netty/NettyUtils.java [hunk 1]
```diff
@@ -43,7 +45,26 @@ public static <T, R, F extends Future<R>> CompletableFuture<T> toCompletableFutu
...
+     * @param <R>         Netty future result type.
+     * @param <F>         Netty future type.
+     * @return CompletableFuture.
+     */
+    public static <T, R, F extends Future<R>> CompletableFuture<T> toCompletableFuture(
+            @Schedule F nettyFuture,
+            Function<F, T> mapper,
+            Supplier<? extends CompletableFuture<T>> completableFutureFactory
+    ) {
+        var fut = completableFutureFactory.get();
```
Comment (SammyVimes) [inline modules/network/src/main/java/org/apache/ignite/internal/network/netty/NettyUtils.java:67]: we can only use `var` for `var smth = new Smth` statements

File: modules/network/src/test/java/org/apache/ignite/network/DefaultMessagingServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+            this.connectionManager = connectionManager;
+            this.messagingService = messagingService;
+        }
+
+        @Override
+        public void close() throws Exception {
+            IgniteUtils.closeAll(connectionManager::stop, messagingService::stop);
+        }
+    }
+}
```
Comment (SammyVimes) [inline modules/network/src/test/java/org/apache/ignite/network/DefaultMessagingServiceTest.java:223]: Missing newline

File: modules/network/src/main/java/org/apache/ignite/internal/network/netty/ConnectionManager.java [hunk 3]
```diff
@@ -92,6 +92,9 @@ public class ConnectionManager {
     /** Recovery descriptor provider. */
     private final RecoveryDescriptorProvider descriptorProvider = new DefaultRecoveryDescriptorProvider();
 
+    /** An action that is invoked before handshake completes. Only useful for tests. */
+    private volatile Runnable beforeHandshakeComplete = () -> {};
```
Comment (SammyVimes) [inline modules/network/src/main/java/org/apache/ignite/internal/network/netty/ConnectionManager.java:96]: I think we should avoid test code in prod code (except for getters, probably). Shall we maybe leverage the factory pattern to create different handshake handler?

**Review by SammyVimes — CHANGES_REQUESTED** [review]

File: modules/network/src/main/java/org/apache/ignite/internal/network/netty/ConnectionManager.java (hunk 3 above)
Comment (rpuch) [reply modules/network/src/main/java/org/apache/ignite/internal/network/netty/ConnectionManager.java:96]: `RecoveryClientHandshakeManager` constructor currently requires 5 arguments. We can pass message factory directly, but we would still need to create a factory with 4 arguments which makes the factory sooo specific. I'm not sure whether it's the best way to go, but I'll try and we'll look at the result.

Comment (rpuch) [reply modules/network/src/main/java/org/apache/ignite/internal/network/netty/ConnectionManager.java]: Done. Does this look botter, WDYT?

**Review by SammyVimes — APPROVED** [review]
LGTM

**Pull request merged**